flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -2046,7 +2046,9 @@ private boolean isValidPojoField(Field f, Class<?> clazz, List<Type> typeHierarc
                 }
                 // check for setters (<FieldName>_$eq for scala)
                 if ((methodNameLow.equals("set" + fieldNameLow)
-                                || methodNameLow.equals(fieldNameLow + "_$eq"))
+                                || methodNameLow.equals(fieldNameLow + "_$eq")
+                                || (fieldNameLow.startsWith("is")
+                                        && methodNameLow.equals("set" + fieldNameLow.substring(2))))
```

Comment:
should there be added a similar check for `has`?
e.g.
```java
boolean hasAncestor = false;
```

Reply:
If that would be a one-liner then I would add that but it would completely blow up the code.
* By default lombok generates `isHasAncestor` and `setHasAncestor` which is just bad
* With `@Accessors(fluent = true)` there is no setter generated + the getter has different signature

Reply:
fair enough